hooks/useERC20Contract.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import ERC20_JSON from 'abi/ERC20.json';
+
+import { useEffect, useState } from 'react';
+
+import Web3 from 'web3';
+import { Contract } from 'web3-eth-contract';
+import { AbiItem } from 'web3-utils';
+
+import PolyjuiceHttpProvider from '@polyjuice-provider/web3';
+
+const providerConfig = {
+  web3Url: 'https://godwoken-testnet-web3-rpc.ckbapp.dev',
+};
+
+const provider = new PolyjuiceHttpProvider(providerConfig.web3Url, providerConfig);
```

Comment:
PolyjuiceHttpProvider and Web3 should be a singleton in whole application. It should be constructed once and passed across the application either in Redux or some shared file.

Reply:
i saved that as TODO to merge that PR now